tools/FakeItEasy.Deploy/Program.cs

```diff
@@ -0,0 +1,104 @@
+            if (string.IsNullOrEmpty(releaseName))
+            {
+                Console.WriteLine("No Appveyor tag name supplied. Not deploying.");
+                return;
+            }
+
+            var (repoOwner, repoName) = GetRepositoryName();
+            var gitHubClient = GetAuthenticatedGitHubClient();
+
+            Console.WriteLine($"Deploying {releaseName}");
+            Console.WriteLine($"Looking for GitHub release {releaseName}");
+
+            var releases = await gitHubClient.Repository.Release.GetAll(repoOwner, repoName);
+            var release = releases.FirstOrDefault(r => r.Name == releaseName)
+                ?? throw new Exception($"Can't find release {releaseName}");
+
+            var artifactsFolder = Path.GetFullPath(Path.Combine(GetCurrentScriptDirectory(), "../../artifacts/output/"));
+            var artifactsPattern = "*.nupkg";
+
+            var artifacts = Directory.GetFiles(artifactsFolder, artifactsPattern);
+            if (!artifacts.Any())
+            {
+                throw new Exception("Can't find any artifacts to publish");
+            }
+
+            Console.WriteLine($"Uploading artifacts to GitHub release {releaseName}");
+            foreach (var file in artifacts)
+            {
+                await UploadArtifactToGitHubReleaseAsync(gitHubClient, release, file);
+            }
+
+            Console.WriteLine("Pushing nupkgs to nuget.org");
+            foreach (var file in artifacts)
+            {
+                await UploadPackageToNuGetAsync(file);
+            }
+
+            Console.WriteLine("Finished deploying");
+        }
+
+        private static async Task UploadArtifactToGitHubReleaseAsync(GitHubClient client, Release release, string path)
+        {
+            var name = Path.GetFileName(path);
+            Console.WriteLine($"Uploading {name}");
+            using (var stream = File.OpenRead(path))
+            {
+                var upload = new ReleaseAssetUpload
+                {
+                    FileName = name,
+                    ContentType = "application/octet-stream",
+                    RawData = stream,
+                    Timeout = TimeSpan.FromSeconds(100)
+                };
+
+                var asset = await client.Repository.Release.UploadAsset(release, upload);
+                Console.WriteLine($"Uploaded {asset.Name}");
+            }
+        }
+
+        private static async Task UploadPackageToNuGetAsync(string path)
+        {
+            const string nugetServer = "https://www.nuget.org/api/v2/package";
+            string name = Path.GetFileName(path);
+            Console.WriteLine($"Pushing {name}");
+            await RunAsync(ToolPaths.NuGet, $"push \"{path}\" -ApiKey {GetNuGetApiKey()} -Source {nugetServer} -NonInteractive -ForceEnglishOutput", noEcho: true);
+            Console.WriteLine($"Pushed {name}");
+        }
+
+        private static (string repoOwner, string repoName) GetRepositoryName()

```

Comment:
:+1: for the multi-return. It's like I'm reading Python.

Reply:
Yeah, I really enjoy tuples too :wink: